packages/react-native/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModule.mm

```diff
@@ -322,7 +365,7 @@ id convertJSIValueToObjCObject(jsi::Runtime &runtime, const jsi::Value &value, s
           resolveWasCalled = YES;
         };
 
-        RCTPromiseRejectBlock rejectBlock = ^(NSString *code, NSString *message, NSError *error) {
+        auto handlePromiseRejection = ^(NSDictionary *reason) {
```

Comment:
Will need rebasing on `https://github.com/facebook/react-native/pull/40792` but should hopefully make this simpler.

Reply:
Since the commit was reverted, I will leave it as is for now.

Reply:
I see it was fixed and now it's in the main branch.